server/lib/chat/chat-models.ts

```diff
@@ -692,17 +693,14 @@ export async function findChannelsByName(
 }
 
 /**
- * Returns a full list of chat channels, optionally filtered by a `searchStr`. Only admins should be
- * able to call this function.
+ * Returns a list of chat channels, optionally filtered by a `searchStr`.
  */
-export async function searchChannelsAsAdmin(
+export async function searchChannels(
   {
-    limit,
-    pageNumber,
+    offset,
```

Comment:
imo you should still have a limit parameter, just specifying the offset and not the "page number"

Really this whole method of pagination is bad (it gets messed up if a channel is ever added/removed), but I'm not gonna make you rewrite *all* of this right now